src/include/libpmemobj/ctl.h

```diff
@@ -46,6 +46,196 @@
+ * class is selected in a fashion that minimizes contention between threads.
+ * Depending on the requested size and the configuration of the size<>class
+ * mapping, it might happen that the object size (including required metadata)
+ * would be bigger than the allocation class size - called unit size. In those
+ * situations, the object is constructed from two or more units (up to 64).
+ *
+ * If the requested number of units cannot be retrieved from the selected
+ * bucket, the thread reaches out to the global, shared, heap which manages
+ * memory in 256 kilobyte chunks and gives it out in a best-fit fashion. This
+ * operation must be performed under an exclusive lock.
+ * Once the thread is in the possesion of a chunk, the lock is dropped, and the
+ * memory is split into units that repopulate the bucket.
+ *
+ * These are the CTL entry points that control allocation classes:
+ * - heap.alloc_class.reset
+ *	Deletes all existing allocation classes and associated resources,
+ *	allowing applications to have complete control over the allocator.
+ *
+ * - heap.alloc_class.[class_id].desc
+ *	Creates/retrieves allocation class information
+ *
+ * - heap.alloc_class.map.range
+ *	Creates a size<>alloc class mapping
+ *
+ * - heap.alloc_class.map.limit
+ *	Retrieves the limit (in bytes) of the map
+ *
+ * - heap.alloc_class.map.granularity
+ *	Retrieves the granularity (in bytes) of the map
+ *
+ * It's VERY important to remember that the allocation classes are a RUNTIME
+ * property of the allocator - they are NOT stored persistently in the pool.
+ * It's recommended to always create custom allocation classes and the
+ * corresponding mapping immediately after creating and opening the pool,
+ * before any use.
+ * If there are existing objects created using a class that is no longer stored
+ * in the runtime state of the allocator, they can be normally freed, but
+ * allocating equivalent objects will be done using the allocation class that
+ * is currently defined for that size.
+ *
+ * Please see the libpmemobj man page for more information.
+ */
+
+/*
+ * Persistent allocation header
+ */
+enum pobj_header_type {
+	/*
+	 * 64-byte header used up until the version 1.3 of the library,
+	 * functionally equivalent to the compact header.
+	 * It's not recommended to create any new classes with this header.
+	 */
+	POBJ_HEADER_LEGACY,
+	/*
+	 * 16-byte header used by the default allocation classes. All library
+	 * metadata is by default allocated using this header.
+	 * Supports type numbers and variably sized allocations.
+	 */
+	POBJ_HEADER_COMPACT,
+	/*
+	 * 0-byte header with metadata stored exclusively in a bitmap. This
+	 * ensures that allocations are allocated in memory contiguously and
+	 * without attached headers.
+	 * This can be used to create very small allocation classes, but it
+	 * does not support type numbers.
+	 * Additionally, allocations with this header can only span a single
+	 * unit.
```

Comment:
It's already technically speaking possible. But it goes way beyond normal use and it breaks the API... You are better off just allocating N single-unit objects in a transaction (after we merge the aggregated tx ops patch, it will perform similarly).